### PR TITLE
[hailtop.utils] Digits needed fresh start

### PR DESCRIPTION
**Pull request by CDiaz96**

### CODE REVIEW

File: hail/python/test/hailtop/utils/test_utils.py [hunk 1]
```diff
@@ -126,3 +128,13 @@ def test_parse_docker_image_reference():
     assert x.digest is None
     assert x.name() == 'us-docker.pkg.dev/my-project/my-repo/test-image'
     assert str(x) == 'us-docker.pkg.dev/my-project/my-repo/test-image'
+
+
+def test_digits_needed():
+    assert digits_needed(0) == 1
+    assert digits_needed(12) == 2
```
Comment (jigold) [inline hail/python/test/hailtop/utils/test_utils.py:135]: Can you add a test for `1` and `3333`?

File: hail/python/test/hailtop/utils/test_utils.py [hunk 2]
```diff
@@ -126,3 +128,13 @@ def test_parse_docker_image_reference():
...
+    assert digits_needed(100) == 3
+    assert digits_needed(3000) == 4
+    assert digits_needed(50000) == 5
+    assert digits_needed(100000) == 6
+
```
Comment (jigold) [inline hail/python/test/hailtop/utils/test_utils.py:140]: Can you add a new line here?

**Review by jigold — DISMISSED** [review]

File: hail/python/test/hailtop/utils/test_utils.py (hunk 2 above)
Comment (CDiaz96) [reply hail/python/test/hailtop/utils/test_utils.py:143]: Added new line

File: hail/python/test/hailtop/utils/test_utils.py (hunk 1 above)
Comment (CDiaz96) [reply hail/python/test/hailtop/utils/test_utils.py:136]: Added new test

File: hail/python/test/hailtop/utils/test_utils.py [hunk 3]
```diff
@@ -1,5 +1,7 @@
 from hailtop.utils import (partition, url_basename, url_join, url_scheme,
                            url_and_params, parse_docker_image_reference)
+from hailtop.utils.utils import digits_needed
+
```
Comment (danking) [inline hail/python/test/hailtop/utils/test_utils.py:4]: Remove this extra line

File: hail/python/test/hailtop/utils/test_utils.py [hunk 4]
```diff
@@ -126,3 +128,16 @@ def test_parse_docker_image_reference():
...
+    assert digits_needed(0) == 1
+    assert digits_needed(1) == 1
+    assert digits_needed(12) == 2
+    assert digits_needed(333) == 3
+    assert digits_needed(100) == 3
+    assert digits_needed(3000) == 4
+    assert digits_needed(50000) == 5
+    assert digits_needed(100000) == 6
+
+
```
Comment (danking) [inline hail/python/test/hailtop/utils/test_utils.py:143]: Remove these two empty lines at the end of the file

**Review by danking — CHANGES_REQUESTED** [review]

**Comment by danking** [comment]
When you’ve made the changes, dismiss my and Jackie’s reviews. If a PR has no review, it appears as “Awaiting Action” for the assigned individuals at https://ci.hail.is/me

File: hail/python/test/hailtop/utils/test_utils.py [hunk 5]
```diff
@@ -1,7 +1,11 @@
 from hailtop.utils import (partition, url_basename, url_join, url_scheme,
                            url_and_params, parse_docker_image_reference, grouped)
+<<<<<<< HEAD
 from hailtop.utils.utils import unzip
+=======
```
Comment (jigold) [inline hail/python/test/hailtop/utils/test_utils.py:5]: You'll need to fix this to combine the two imports on one line and remove the lines that start with `====` and `<<<` and `>>>`.

```
from hailtop.utils.utils import unzip, digits_needed
```

**Review by jigold — CHANGES_REQUESTED** [review]

File: hail/python/test/hailtop/utils/test_utils.py (hunk 5 above)
Comment (CDiaz96) [reply hail/python/test/hailtop/utils/test_utils.py]: Fixed

**Pull request merged**